File: 卡码/代码随想录/Java/src/com/kurumi/codeThoughts/栈与队列/_01_有效括号/LeetCode_0020.java
package com.kurumi.codeThoughts.栈与队列._01_有效括号;

import java.util.Stack;

/**
 * @Author kurumi
 * @Date 2026-07-03 上午11:52:20
 * @Description 20. 有效的括号
 */
public class LeetCode_0020 {
    static class Solution {
        public boolean isValid(String s) {
            if (s.length() % 2 != 0)
                return false;
            Stack<Character> stack = new Stack<>();
            for (int i = 0; i < s.length(); i++) {
                if (s.charAt(i) == '(')
                    stack.push(')');
                else if (s.charAt(i) == '[')
                    stack.push(']');
                else if (s.charAt(i) == '{')
                    stack.push('}');
                else if (stack.empty() || stack.peek() != s.charAt(i))
                    return false;
                else
                    stack.pop();
            }
            return stack.empty();
        }
    }
}

File: 卡码/代码随想录/JS/栈与队列/01-有效括号/LeetCode_0020.js
//20. 有效的括号
const isValid = function(s) {
  if(s.length%2!=0) return false;
  let stack=new Array();
  for(let ch of s){
    if(ch=="(") stack.push(")");
    else if(ch=="[") stack.push("]");
    else if(ch=="{") stack.push("}");
    else if(stack.length==0||stack[stack.length-1]!=ch) return false;
    else stack.pop();
  }
  return stack.length==0;
};